drivers/ma_reports/app/models/ma_reports/csg_engage/concerns/has_report_status.rb

```diff
@@ -0,0 +1,45 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module MaReports::CsgEngage::Concerns::HasReportStatus
+  extend ActiveSupport::Concern
+  included do
+    [
+      :completed,
+      :failed,
+      :in_progress,
+      :created,
+      :unknown,
+    ].each do |status_value|
+      define_method(:"#{status_value}?") { status == status_value }
+    end
+
+    def status
+      return :completed if completed_at.present?
```

Comment:
You may want to add `started_at < 1.days.ago` (and no `completed_at`) as a failure as well.  That should catch the situation it just doesn't complete for whatever reason but doesn't actually fail.